lib/acme/client/faraday_middleware.rb

```diff
@@ -67,14 +75,14 @@ def error_name
   end
 
   def error_type_to_klass(type)
-    type.gsub('urn:acme:error:', '').split(/[_-]/).map { |type_part| type_part[0].upcase + type_part[1..-1] }.join
+    type.gsub('urn:ietf:params:acme:error:', '').split(/[_-]/).map { |type_part| type_part[0].upcase + type_part[1..-1] }.join
```

Comment:
1. I feel like this logic should `live in lib/acme/client/error.rb`
2. A simple `case` on whatever comes after `urn:ietf:params:acme:error:` would be much simpler to reason about (and would let us have less ambiguous error class names https://github.com/unixcharles/acme-client/pull/134/files#r180874585)
3. If you're against (2), 
    - `gsub` &rarr; `sub` would make it clear there's only one substitution going on
    - `sub(/\Aurn:ietf:params:acme:error:/)` is a bit more intention-revealing seen as `urn:ietf:params:acme:error:` should always be at the beginning

Reply:
Good point. When I came up with this there was no error list in the spec. So I just `#constantize` the error message and kinda defined errors has they appear. I'll replace this with a simple hash in the Error module.